File: Java/src/main/java/OneLang/One/CompilerHelper/CompilerHelper.java
package OneLang.One.CompilerHelper;

import OneStd.OneFile;
import OneLang.One.Compiler.Compiler;

import OneLang.One.Compiler.Compiler;
import OneStd.Objects;
import java.util.Arrays;

public class CompilerHelper {
    public static String baseDir = "./";
    
    public static Compiler initProject(String projectName, String sourceDir, String lang, String packagesDir) {
        if (!Objects.equals(lang, "ts"))
            throw new Error("Only typescript is supported.");
        
        var compiler = new Compiler();
        var result = packagesDir;
        compiler.init(result != null ? result : CompilerHelper.baseDir + "packages/");
        compiler.setupNativeResolver(OneFile.readText(CompilerHelper.baseDir + "langs/NativeResolvers/typescript.ts"));
        compiler.newWorkspace(projectName);
        
        for (var file : Arrays.stream(OneFile.listFiles(sourceDir, true)).filter(x -> x.endsWith(".ts")).toArray(String[]::new))
            compiler.addProjectFile(file, OneFile.readText(sourceDir + "/" + file));
        
        return compiler;
    }
}

File: Python/OneLang/One/CompilerHelper.py
from OneLangStdLib import *
from OneFile import *
import OneLang.One.Compiler as comp
import OneLangStdLib as one

@one.static_init
class CompilerHelper:
    @classmethod
    def static_init(cls):
        cls.base_dir = "./"
    
    def __init__(self):
        pass
    
    @classmethod
    def init_project(cls, project_name, source_dir, lang = "ts", packages_dir = None):
        if lang != "ts":
            raise Error("Only typescript is supported.")
        
        compiler = comp.Compiler()
        compiler.init(packages_dir or f'''{CompilerHelper.base_dir}packages/''')
        compiler.setup_native_resolver(OneFile.read_text(f'''{CompilerHelper.base_dir}langs/NativeResolvers/typescript.ts'''))
        compiler.new_workspace(project_name)
        
        for file in list(filter(lambda x: x.endswith(".ts"), OneFile.list_files(source_dir, True))):
            compiler.add_project_file(file, OneFile.read_text(f'''{source_dir}/{file}'''))
        
        return compiler